color/src/colorspace.rs

```diff
@@ -421,6 +421,110 @@ impl ColorSpace for ProphotoRgb {
     }
 }
 
+/// 🌌 The Rec. 2020 color space.
+///
+/// Rec. 2020 is similar to [sRGB](`Srgb`) but has higher red, green and blue chromaticities,
+/// thereby extending its gamut over sRGB on all components.
+///
+/// Its components are `[r, g, b]` (red, green, and blue channels respectively), with `[0, 0, 0]`
+/// pure black and `[1, 1, 1]` white. The natural bounds of the channels are `[0, 1]`.
+///
+/// This corresponds to the color space in [CSS Color Module Level 4 § 10.7][css-sec] and is
+/// [characterized by the ICC][icc]. The color space is defined by the International
+/// Telecommunication Union [here][itu].
+///
+/// [css-sec]: https://www.w3.org/TR/css-color-4/#predefined-rec2020
+/// [icc]: https://www.color.org/chardata/rgb/BT2020.xalter
+/// [itu]: https://www.itu.int/rec/R-REC-BT.2020/en
+#[derive(Clone, Copy, Debug)]
+pub struct Rec2020;
+
+impl Rec2020 {
+    // These are the parameters of the transfer function defined in the Rec. 2020 specification.
+    // They are truncated here to f32 precision.
+    const A: f32 = 1.099_296_8;
+    const B: f32 = 0.018_053_97;
+}
+
+impl ColorSpace for Rec2020 {
+    const TAG: Option<ColorSpaceTag> = Some(ColorSpaceTag::Rec2020);
+
+    fn to_linear_srgb([r, g, b]: [f32; 3]) -> [f32; 3] {
+        // XYZ_to_lin_sRGB * lin_Rec2020_to_XYZ
+        #[expect(
+            clippy::cast_possible_truncation,
+            reason = "exact rational, truncate at compile-time"
+        )]
+        const LINEAR_REC2020_TO_SRGB: [[f32; 3]; 3] = [
+            [
+                (2_785_571_537. / 1_677_558_947.) as f32,
+                (-985_802_650. / 1_677_558_947.) as f32,
+                (-122_209_940. / 1_677_558_947.) as f32,
+            ],
+            [
+                (-4_638_020_506. / 37_238_079_773.) as f32,
+                (42_187_016_744. / 37_238_079_773.) as f32,
+                (-310_916_465. / 37_238_079_773.) as f32,
+            ],
+            [
+                (-97_469_024. / 5_369_968_309.) as f32,
+                (-3_780_738_464. / 37_589_778_163.) as f32,
+                (42_052_799_795. / 37_589_778_163.) as f32,
```

Comment:
Some more lily-gilding here. Debatable usefulness, as the transfer function parameters are reals not representable as rationals.